src/core/speechrecognition/locales.js

```diff
@@ -0,0 +1,25 @@
+import { SPEECH_RECOGNITION_LOCALES_SUPPORTED_BY_EDGE } from '../constants';
+
+/**
+ * Transforms the given locale to a locale Microsoft Edge can understand.
+ * This means changing the language/locale separating underscore to a dash,
+ * and defaulting the locale to the 2 character language code if it is not
+ * supported by Edge.
+ *
+ * Expects the input locale to use an underscore separator as opposed to a dash
+ * separator.
+ *
+ * @param {string} locale
+ * @returns {string}
+ */
+export function transformSpeechRecognitionLocaleForEdge (locale) {
+  const underscoreIndex = locale.indexOf('_');
+  if (underscoreIndex === -1) {
+    return locale;
+  }
+  const isCompatibleWithEdge = SPEECH_RECOGNITION_LOCALES_SUPPORTED_BY_EDGE.includes(locale.toLowerCase());
```

Comment:
Is is possible that the supplied locale has a dash and therefore it wouldn't be found in the list of supported locales?

Reply:
Yeah, but that's only possible if it's a user supplied locale. That's why I added the ` Expects the input locale to use an underscore separator as opposed to a dash separator.` part of the jsdoc. I can add handling if you think it's necessary, I was just a little worried about locales that have multiple separators.

Reply:
I don't think we deal with any locales like that to be fair so I'll just add the handling